src/lib.rs

```diff
@@ -301,6 +301,176 @@ pub unsafe fn replace_with_or_abort_unchecked<T, F: FnOnce(T) -> T>(dest: &mut T
+/// is then placed at the original value's location.
+///
+/// This is effectively the same function as [`replace_with_or_abort`], but it lets the closure
+/// return a custom value as well.
+///
+/// # An important note
+///
+/// On panic (or to be more precise, unwinding) of the closure `f`, the process will **abort** to
+/// avoid returning control while `dest` is in a potentially invalid state.
+///
+/// If this behaviour is undesirable, use [`replace_with_and_return`] or
+/// [`replace_with_or_default_and_return`] instead.
+///
+/// Equivalent to `replace_with_and_return(dest, || process::abort(), f)`.
+///
+/// # Example
+///
+/// ```
+/// # use replace_with::*;
+///
+/// fn take<T>(option: &mut Option<T>) -> Option<T> {
+/// 	replace_with_or_abort_and_return(option, |option| (option, None))
+/// }
+/// ```
+#[inline]
+#[cfg(feature = "std")]
+pub fn replace_with_or_abort_and_return<T, U, F: FnOnce(T) -> (U, T)>(dest: &mut T, f: F) -> U {
+	replace_with_and_return(dest, || std::process::abort(), f)
+}
+
+#[inline]
+#[cfg(all(not(feature = "std"), feature = "nightly"))]
+pub fn replace_with_or_abort_and_return<T, U, F: FnOnce(T) -> (U, T)>(dest: &mut T, f: F) -> U {
+	replace_with_and_return(dest, || unsafe { std::intrinsics::abort() }, f)
+}
+
+/// Temporarily takes ownership of a value at a mutable location, and replace it with a new value
+/// based on the old one. Aborts on panic. Lets the closure return a custom value as well.
+///
+/// We move out of the reference temporarily, to apply a closure `f`, returning a new value, which
+/// is then placed at the original value's location.
+///
+/// This is effectively the same function as [`replace_with_or_abort_unchecked`], but it lets the
+/// closure return a custom value as well.
+///
+/// # An important note
+///
+/// On panic (or to be more precise, unwinding) of the closure `f`, the process will **abort** to
+/// avoid returning control while `dest` is in a potentially invalid state.
+///
+/// Unlike for `replace_with_or_abort()` users of `replace_with_or_abort_unchecked()` are expected
+/// to have `features = ["panic_abort", …]` defined in `Cargo.toml`
+/// and `panic = "abort"` defined in their profile for it to behave semantically correct:
+///
+/// ```toml
+/// # Cargo.toml
+///
+/// [profile.debug]
+/// panic = "abort"
+///
+/// [profile.release]
+/// panic = "abort"
+/// ```
+///
+/// **Word of caution:** It is crucial to only ever use this function having defined `panic = "abort"`,
+/// or else bad things may happen. It's *up to you* to uphold this invariant!
+///
+/// If this behaviour is undesirable, use [`replace_with_and_return`] or
+/// [`replace_with_or_default_and_return`].
+///
+/// Equivalent to `replace_with_and_return(dest, || process::abort(), f)`.
+///
+/// # Example
+///
+/// ```
+/// # use replace_with::*;
+///
+/// fn unsafe take<T>(option: &mut Option<T>) -> Option<T> {
```

Comment:
This is what's failing the build:
```suggestion
/// unsafe fn take<T>(option: &mut Option<T>) -> Option<T> {
```